src/solidity/decodeInfo.js

```diff
@@ -0,0 +1,353 @@
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function Bool (type) {
+  return {
+    needsFreeStorageSlot: false,
+    storageBytes: 1,
+    typeName: 'bool',
+    name: 'bool'
+  }
+}
+
+/**
+  * DynamicByteArray decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function DynamicByteArray (type) {
+  return {
+    needsFreeStorageSlot: true,
+    storageBytes: 32,
+    typeName: 'bytes',
+    name: 'dynamicByteArray'
+  }
+}
+
+/**
+  * FixedByteArray decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function FixedByteArray (type) {
+  if (type.split(' ')) {
+    type = type.split(' ')[0]
+  }
+  return {
+    needsFreeStorageSlot: false,
+    storageBytes: parseInt(type.replace('bytes', '')),
+    typeName: type.split(' ')[0],
+    name: 'fixedByteArray'
+  }
+}
+
+/**
+  * Int decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function Int (type) {
+  if (type.split(' ')) {
+    type = type.split(' ')[0]
+  }
+  return {
+    needsFreeStorageSlot: false,
+    storageBytes: parseInt(type.replace('int', '')) / 8,
+    typeName: type,
+    name: 'int'
+  }
+}
+
+/**
+  * StringType decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function StringType (type) {
+  return {
+    needsFreeStorageSlot: true,
+    storageBytes: 32,
+    typeName: 'string',
+    name: 'stringType'
+  }
+}
+
+/**
+  * ArrayType decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder, arraySize, subArray}
+  */
+function ArrayType (type, stateDefinitions) {
+  var arraySize
+  var storageBytes
+
+  var underlyingType = extractUnderlyingType(type)
+
+  arraySize = extractArraySize(type)
+  var dimensions = extractDimensions(type)
+  type = underlyingType + dimensions.join('')
+
+  var subArrayType = type.substring(0, type.lastIndexOf('['))
+  var subArray = null
+  if (subArrayType.indexOf('[') !== -1) {
+    subArray = decode(subArrayType, stateDefinitions)
+  }
+
+  underlyingType = decode(underlyingType, stateDefinitions)
+  storageBytes = underlyingType.storageBytes
+
+  if (arraySize === 'dynamic') {
+    storageBytes = 32
+  } else {
+    if (subArray) {
```

Comment:
You have to use underlyingType.storageBytes and calculate how many elements of the underlying type will fit into 32 bytes (or how many slots a single element will occupy). Then use that to compute the size.

Oh and please take into account that especially with arrays, the size can easily overflow the precision of javascript numbers, so be prepared to create some kind of error.